niworkflows/interfaces/bids.py

```diff
@@ -464,34 +466,45 @@ def _run_interface(self, runtime):
             self._results['out_file'].append(out_file)
             self._results['compression'].append(_copy_any(fname, out_file))
 
-            is_nii = out_file.endswith('.nii') or out_file.endswith('.nii.gz')
-            if self.inputs.check_hdr and is_nii:
+            is_nii = out_file.endswith(('.nii', '.nii.gz'))
+            if is_nii and any((self.inputs.check_hdr, self.inputs.data_dtype)):
```

Comment:
```suggestion
            is_nii = out_file.endswith(('.nii', '.nii.gz'))
            data_dtype = DEFAULT_DTYPES[self.inputs.suffix]
            if isdefined(self.inputs.data_dtype):
                data_dtype = self.inputs.data_dtype
            if is_nii and any((self.inputs.check_hdr, self.inputs.data_dtype)):
```

where
```Python
from collections import defaultdict

def _none():
     return None

DEFAULT_DTYPES = defaultdict({
    "mask": "uint8",
    "dseg": "int16",
    "probseg": "float32",
}, _none)
```

Reply:
i've incorporated this, but would like to still include the input field `data_dtype` to allow overriding this JIC

Reply:
@oesteban did the latest commit address your concerns?